src/utils/sourcemaps.rs

```diff
@@ -852,7 +853,19 @@ impl SourceMapProcessor {
         } else {
             println!("{} Nothing to upload", style(">").dim());
         }
-        Ok(())
+        Ok(files_needing_upload)
+    }
+
+    /// Upload all files in "strict" mode. Strict mode differs from a normal upload
+    /// only when there are no files to upload. In strict mode, having no files to
+    /// upload results in an error, whereas such an upload is successful in normal
+    /// (non-strict) mode. On success, the number of uploaded files is returned in
+    /// an Ok()
+    pub fn upload_strict(&mut self, context: &UploadContext<'_>) -> Result<usize> {
```

Comment:
I implemented this change by creating a separate function for strict uploads. The other options I considered were adding a field to the `UploadContext` struct or an argument to the `upload` function which would store a boolean representing whether strict mode is enabled. However, since we call the `upload` function and use the `UploadContext` struct in multiple places, we would need to change all of these uses to go with either or those options.

Reply:
I think it makes perfect sense to add a separate function for this.